src/EnergyPlus/RuntimeLanguageProcessor.cc

```diff
@@ -2587,6 +2587,30 @@ namespace RuntimeLanguageProcessor {
         }
     }
 
+    int TodayTomorrowWeather(int hour, int timestep, Array2D<Real64> &TodayTomorrowWeatherSource, Real64 &value) {
+        int iHour = hour + 1;
+        if ((iHour > 0) && (iHour <= 24) && (timestep > 0) && (timestep <= DataGlobals::NumOfTimeStepInHour)) {
+            value = TodayTomorrowWeatherSource(timestep, iHour);
+            return 0;
+        } else {
+            return 1;
+        }
+    }
+
+    int TodayTomorrowWeather(int hour, int timestep, Array2D<bool> &TodayTomorrowWeatherSource, int &value) {
+        int iHour = hour + 1;
+        if ((iHour > 0) && (iHour <= 24) && (timestep > 0) && (timestep <= DataGlobals::NumOfTimeStepInHour)) {
+            if (TodayTomorrowWeatherSource(timestep, iHour)) {
+                value = 1.0;
+            } else {
+                value = 0.0;
+            }
+            return 0;
+        } else {
+            return 1;
+        }
+    }
```

Comment:
I added my own worker functions for mining the specific weather data.  It's not 100% ideal, but it's like 90% ideal.  It's still really nice to pack this logic into reusable pieces.  I couldn't use the ones previously added because their job was specifically to update Erl operands, and I needed raw values.  No time to make it more elegant right now.